monai/utils/profiling.py

```diff
@@ -102,9 +125,294 @@ def __init__(self):
         self.start_time = None
 
     def __enter__(self):
-        self.start_time = time.perf_counter()
+        self.start_time = perf_counter()
         return self
 
     def __exit__(self, exc_type, exc_value, exc_traceback):
-        self.total_time += time.perf_counter() - self.start_time
+        self.total_time += perf_counter() - self.start_time
         self.start_time = None
+
+
+# stores the results from profiling with trace or with other helper methods
+ProfileResult = namedtuple("ProfileResult", ["name", "time", "filename", "lineno", "pid", "timestamp"])
+
+
+def select_transform_call(frame):
+    """Returns True if `frame` is a call to a `Transform` object's `_call__` method."""
+    from monai.transforms import Transform  # prevents circular import
+
+    self_obj = frame.f_locals.get("self", None)
+    return frame.f_code.co_name == "__call__" and isinstance(self_obj, Transform)
+
+
+class WorkflowProfiler:
+    """
+    Profiler for timing all aspects of a workflow. This includes using stack tracing to capture call times for
+    all selected calls (by default calls to `Transform.__call__` methods), times within context blocks, times
+    to generate items from iterables, and times to execute decorated functions.
+
+    This profiler must be used only within its context because it uses an internal thread to read results from a
+    multiprocessing queue, this allows the profiler to function across multiple threads and processes. The
+    profiler uses `sys.settrace` and `threading.settrace` to find all calls to profile, this will be set when
+    the context enters and cleared when it exits so proper use of the context is essential to prevent excessive
+    tracing. Note that tracing has a high overhead so times will not accurately reflect real world performance
+    but give an idea of relative share of time spent.
+
+    The tracing functionality uses a selector to choose which calls to trace, since tracing all calls induces
+    infinite loops and would be terribly slow even if not. This selector is a callable accepting a `call` trace
+    frame and returns True if the call should be traced. The dedault is `select_transform_call` which will return
+    True for `Transform.__call__` calls only.
+
+    Example showing use of all profiling functions:
+
+    .. code-block:: python
+
+        import monai.transform as mt
+        comp=mt.Compose([mt.ScaleIntensity(),mt.RandAxisFlip(0.5)])
```

Comment:
cool example, it works fine but requires these imports as I test it:
```py
import monai.transforms as mt
from monai.utils import WorkflowProfiler
import torch
```

Reply:
Added in.